ColorChooser Color Wheel

ericmehl opened the pull request:
This changes the type of color field to a color wheel instead of the rectangular field when hue is being visualized. So it's active when the static color is set to saturation or value.

For drawing the field, the 50 pixel bitmap with bilinear scaling up to the widget size gives pretty good performance and doesn't introduce too much inaccuracy compared to drawing the field at the full resolution. But as we discussed yesterday, it probably could be much better if we did the drawing in GLSL or c++, which is probably worth investigating as a follow-up.

### Checklist ###

- [X] I have read the [contribution guidelines](https://github.com/GafferHQ/gaffer/blob/main/CONTRIBUTING.md).
- [X] I have updated the documentation, if applicable.
- [X] I have tested my change(s) in the test suite, and added new test cases where necessary.
- [X] My code follows the Gaffer project's prevailing coding style and conventions.

murraystevenson reviewed (changes requested):
Thanks Eric! I took this for a spin in the context of #6003 and had a couple of thoughts.

- We'll need to ignore button presses outside of the wheel. Currently you can click in the square surrounding the wheel and generate out of range values.
- With the wheel in place, do you think "v" would be the better default static component? Currently we have "h" here and "s" over in #6003.
- Unrelated to the wheel, but a general note for the ColorField: we'll need to implement `_ColorField._displayTransformChanged()` along the lines of [this](https://github.com/GafferHQ/gaffer/blob/6a30266bd58ff8edbade31daa2dbe5b788906877/python/GafferUI/ColorChooser.py#L161-L164) so we redraw the field on display transform updates. At present the sliders respond to changes to `openColorIO.displayTransform` in the script settings, but the field does not until you manually cause it to redraw.

murraystevenson commented:
> * Unrelated to the wheel, but a general note for the ColorField: we'll need to implement `_ColorField._displayTransformChanged()`...

John has addressed this in https://github.com/GafferHQ/gaffer/pull/6080/commits/d01ce8f62bf30563baaf3c01251418bdc21a40a9 as part of #6080.

ericmehl commented on python/GafferUI/ColorChooser.py:
```diff
@@ -450,8 +504,15 @@ def __paintEvent( self, event ) :
+
 		self.__drawBackground( painter )
 
+		painter.setClipping( False )
```
Fixed in https://github.com/GafferHQ/gaffer/pull/5999/commits/f4a6645d6c7b74eb2290b322e46bf5b1a7a547c6

ericmehl commented:
Thanks for the review @murraystevenson!

> * We'll need to ignore button presses outside of the wheel. Currently you can click in the square surrounding the wheel and generate out of range values.

Good catch, fixed in https://github.com/GafferHQ/gaffer/pull/5999/commits/7f3eae2d2965bc892b10e9b7de8cb1408342b007

> * With the wheel in place, do you think "v" would be the better default static component? Currently we have "h" here and "s" over in

Noting here the conversation we had offline - we decided to leave the default fixed component as "s", which is the default in #6003. It results in a grayscale wheel going from black in the center to white at the edge, which is perhaps not the most helpful. But soon we're going to be adding the ability to make the sliders and wheel / field either dynamic (as they are now) or static (colors don't change based on the current color). When that is in place, the default wheel will have full saturation make more sense.

With the fix in #6080, the current comments are addressed. Ready for a new look!

murraystevenson commented:
Thanks Eric! I think my preference would still be for "v" to be static by default, but users can configure that if they have a similar inclination...

Think we'll just need to include a changelog entry in case we release another 1.5 alpha, though this could be tagged with the `[^1]` footnote as it's not likely necessary in the combined 1.5.0.0 release notes. Feel free to do this while squashing this down ready to merge!

ericmehl commented:
Thanks Murray! I squashed down and added a changelog entry with the `[^1]` notation. And rebased on `main`, so hopefully we get some more happy green checks than we've been getting lately.
